app/validators/blocks/calculated_summary_block_validator.py

```diff
@@ -49,3 +57,35 @@ def validate(self):
             ):
                 self.add_error(self.ANSWERS_MUST_HAVE_SAME_CURRENCY)
         return self.errors
+
+    def validate_answer_id_set_before_calculated_summary_block(self):
+        for answer_id in self.answers_to_calculate:
+            answer_id_block = self.questionnaire_schema.get_block_by_answer_id(
+                answer_id
+            )
+            if self.questionnaire_schema.block_ids.index(
+                answer_id_block["id"]
+            ) > self.questionnaire_schema.block_ids.index(self.block["id"]):
+                self.add_error(
+                    self.ANSWER_SET_AFTER_CALCULATED_SUMMARY, block=self.block
+                )
+
+    def validate_answer_id_for_calculated_summary_not_in_different_section(self):
+        answer_section_ids = {
+            self.questionnaire_schema.answers_with_context[answer_id]["section"]
+            for answer_id in self.answers_to_calculate
+        }
+        if len(answer_section_ids) > 1:
+            self.add_error(
+                self.ANSWER_SET_IN_DIFFERENT_SECTION_FOR_CALCULATED_SUMMARY,
+                block=self.block,
+            )
+
+    def get_calculated_answer_ids(self):
+        if self.block["calculation"].get("answers_to_calculate"):
+            return self.block["calculation"]["answers_to_calculate"]
+        return [
+            value["identifier"]
+            for value in self.block["calculation"]["operation"]["+"]
```

Comment:
We shouldn't expect it to always be one level. `+` will allow recursion, so this should get all identifier at all levels. I think we must have a function to return all values for identifiers for a given schema such as `self.block["calculation"]["operation"]`

Reply:
Added the check